omaha/goopdate/download_manager.cc

```diff
@@ -552,6 +565,22 @@ HRESULT DownloadManager::CachePackage(const Package* package,
   return hr;
 }
 
+HRESULT DownloadManager::EnsureSignatureIsValid(const CString& file_path) {
+  const TCHAR* ext = ::PathFindExtension(file_path);
+  ASSERT1(ext);
+  if (*ext != _T('\0')) {
+    ext++;  // Skip the dot.
+    CString ext_lower(ext);
+    ext_lower.MakeLower();
+    const TCHAR* delim = _T(" ");
+    CString extensions = delim + CString(kAuthenticodeSignedExtensions) + delim;
+    if (extensions.Find(delim + ext_lower + delim) != -1) {
```

Comment:
I suggest using a set or a vector and a find algorithm to look up the extension instead of string matching.

Reply:
I committed an implementation in bb826fe14b76fe1cf731175a928974a2a97dbbe2 that uses a char array. I didn't know how you would want me to use a set or vector. Using a set or vector would add them as a dependency in `constants.h` and that felt too far-reaching. Having a char array in `constants.h` then turning it into a set or vector in `download_manager.cc` felt overly complicated. Would love to hear your thoughts.

Reply:
Your code would work. The array is short and these extensions are looked up infrequently. The ideas suggested were:
* using a set of strings to look them the extension up
* using some kind of container which can be traversed using a find algo. Your original code is using a Find too, but it is string matching Find, so we need to be careful about the `delim`, boundary cases, etc.